File: CS/TestConditionalAppearanceRules.UnitTests/FakeAppearanceItem.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Drawing;
using DevExpress.ExpressApp.Editors;

namespace TestConditionalAppearanceRules.UnitTests {
    public class FakeAppearanceTarget : 
        IAppearanceFormat, IAppearanceEnabled, IAppearanceVisibility {
        #region IAppearanceFormat Members
        private Color backColor;
        public Color BackColor { 
            get { return backColor; } 
            set { backColor = value; } 
        }
        private Color fontColor;
        public Color FontColor { 
            get { return fontColor; } 
            set { fontColor = value; } 
        }
        private FontStyle fontstyle;
        public FontStyle FontStyle { 
            get { return FontStyle; } 
            set { fontstyle = value; } 
        }
        public void ResetFontStyle() {
            FontStyle = FontStyle.Regular;
        }
        public void ResetFontColor() {
            FontColor = new Color();
        }
        public void ResetBackColor() {
            FontColor = new Color();
        }
        #endregion
        #region IAppearanceEnabled Members
        private bool enabled;
        public bool Enabled { 
            get { return enabled; } 
            set { enabled = value; } 
        }
        public void ResetEnabled() {
            Enabled = true;
        }
        #endregion
        #region IAppearanceVisibility Members
        private ViewItemVisibility visibility = ViewItemVisibility.Show;
        public ViewItemVisibility Visibility {
            get { return visibility; }
            set {
                if (visibility != value) {
                    visibility = value;
                }
            }
        }
        public void ResetVisibility() {
            Visibility = ViewItemVisibility.Show;
        }
        #endregion
    }
}

File: VB/TestConditionalAppearanceRules.UnitTests/FakeAppearanceItem.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Drawing
Imports DevExpress.ExpressApp.Editors

Namespace TestConditionalAppearanceRules.UnitTests
    Public Class FakeAppearanceTarget
        Implements IAppearanceFormat, IAppearanceEnabled, IAppearanceVisibility

        #Region "IAppearanceFormat Members"

        Private _backColor As Color
        Public Property BackColor() As Color Implements IAppearanceFormat.BackColor
            Get
                Return _backColor
            End Get
            Set(ByVal value As Color)
                _backColor = value
            End Set
        End Property

        Private _fontColor As Color
        Public Property FontColor() As Color Implements IAppearanceFormat.FontColor
            Get
                Return _fontColor
            End Get
            Set(ByVal value As Color)
                _fontColor = value
            End Set
        End Property

        Private _fontstyle As FontStyle
        Public Property FontStyle() As FontStyle Implements IAppearanceFormat.FontStyle
            Get
                Return _fontstyle
            End Get
            Set(ByVal value As FontStyle)
                _fontstyle = value
            End Set
        End Property
        Public Sub ResetFontStyle() Implements IAppearanceFormat.ResetFontStyle
            FontStyle = FontStyle.Regular
        End Sub
        Public Sub ResetFontColor() Implements IAppearanceFormat.ResetFontColor
            FontColor = New Color()
        End Sub
        Public Sub ResetBackColor() Implements IAppearanceFormat.ResetBackColor
            FontColor = New Color()
        End Sub
        #End Region
        #Region "IAppearanceEnabled Members"

        Private _enabled As Boolean
        Public Property Enabled() As Boolean Implements IAppearanceEnabled.Enabled
            Get
                Return _enabled
            End Get
            Set(ByVal value As Boolean)
                _enabled = value
            End Set
        End Property
        Public Sub ResetEnabled() Implements IAppearanceEnabled.ResetEnabled
            Enabled = True
        End Sub
#End Region

        #Region "IAppearanceVisibility Members"
        Private _visibility As ViewItemVisibility = ViewItemVisibility.Show
        Public Property Visibility() As ViewItemVisibility Implements IAppearanceVisibility.Visibility
            Get
                Return _visibility
            End Get
            Set(ByVal value As ViewItemVisibility)
                If (Not _visibility.Equals(value)) Then
                    _visibility = value
                End If
            End Set
        End Property
        Public Sub ResetVisibility() Implements IAppearanceVisibility.ResetVisibility
            Visibility = ViewItemVisibility.Show
        End Sub
        #End Region
    End Class
End Namespace
